packages/vow/test/watch-upgrade.test.js

```diff
@@ -0,0 +1,63 @@
+import {
+  annihilate,
+  startLife,
+  test,
+} from '@agoric/swingset-vat/tools/prepare-strict-test-env-ava.js';
+
+import { makeDurableZone } from '@agoric/zone/durable.js';
+
+import { prepareVowTools } from '../vat.js';
+
+test.serial('vow resolve across upgrade', async t => {
+  annihilate();
+
+  t.plan(1);
+
+  await startLife(async baggage => {
+    const zone = makeDurableZone(baggage, 'durableRoot');
+    const vowTools = prepareVowTools(zone);
+    const watcher = zone.exo('DurableVowTestWatcher', undefined, {
+      onFulfilled(value) {
+        t.fail(
+          `First incarnation watcher onFulfilled triggered with value ${value}`,
+        );
+      },
+      onRejected(reason) {
+        t.fail(
+          `First incarnation watcher onRejected triggered with reason ${reason}`,
+        );
+      },
+    });
+
+    const testVowKit = zone.makeOnce('testVowKit', () => vowTools.makeVowKit());
+
+    vowTools.watch(testVowKit.vow, watcher);
+  });
+
+  await startLife(
+    baggage => {
+      const zone = makeDurableZone(baggage, 'durableRoot');
+      prepareVowTools(zone);
+      zone.exo('DurableVowTestWatcher', undefined, {
```

Comment:
Did you mean to do something with this watcher?  It seems just to be dropped on the floor.

```suggestion
      const watcher = zone.exo('DurableVowTestWatcher', undefined, {
```

Reply:
The behavior is redefined. This singleton exo is already setup as watcher in the previous incarnation, so we don't care about its identity here, just that a behavior is reconnected when invoked when resolving the vow.
Added a comment.